### ISSUE TITLE
Remove hardcoded enum values from the frontend

### ISSUE THREAD

**Issue by frozenhelium** [IFRCGo/go-frontend#1529]
#### Problem
Currently we have to maintain  the enum values such as Disaster types, Organization types, ERU types, etc, in both frontend and backend. We should only have one source of truth and one place to maintain such values. eg: for translation 

#### Solution
We should fetch all of these enums from the server in the initialization phase.

The hardcoded enum values are defined in the following files
* [`utils/field-report-constants.js`](https://github.com/IFRCGo/go-frontend/blob/develop/src/root/utils/field-report-constants.js)
* [`utils/appeal-type-constants.js`](https://github.com/IFRCGo/go-frontend/blob/develop/src/root/utils/appeal-type-constants.js)
* [`utils/constants.js`](https://github.com/IFRCGo/go-frontend/blob/develop/src/root/utils/constants.js)
* [`utils/eru-types.js`](https://github.com/IFRCGo/go-frontend/blob/develop/src/root/utils/eru-types.js)
* [`utils/region-constants`](https://github.com/IFRCGo/go-frontend/blob/develop/src/root/utils/region-constants.js)

**Comment by nanometrenat** [comment]
I think this is a duplicate of https://github.com/IFRCGo/go-frontend/issues/940 can you please check and confirm? @batpad maybe you have advice on best way to split the tickets if needed!

Noting that there are also several other separate tickets about specific enums (e.g. geospatial related ones) that have been (or are in) the process of being addressed

**Comment by frozenhelium** [comment]
Depends on: https://github.com/IFRCGo/go-api/issues/889

**Comment by frozenhelium** [comment]
@nanometrenat It's similar to the issue you mentioned above, but not exactly.  https://github.com/IFRCGo/go-frontend/issues/940 is about the model fields and this issue is specifically about the ENUM / constant fields

**Comment by nanometrenat** [comment]
@frozenhelium @batpad @tovari did this get addressed in the GO rewrite or is it still an open thing for build?  Cheers
